src/libraries/System.Reflection.Metadata/tests/Metadata/Ecma335/Encoding/InstructionEncoderTests.cs

```diff
@@ -182,16 +182,20 @@ public void LoadConstantR4()
             il.LoadConstantR4(float.PositiveInfinity);
             il.LoadConstantR4(float.Epsilon);
 
-            AssertEx.Equal(new byte[]
+            var expectedEncoding = new byte[]
             {
                 (byte)ILOpCode.Ldc_r4, 0x00, 0x00, 0x00, 0x00,
                 (byte)ILOpCode.Ldc_r4, 0xFF, 0xFF, 0x7F, 0x7F,
                 (byte)ILOpCode.Ldc_r4, 0xFF, 0xFF, 0x7F, 0xFF,
-                (byte)ILOpCode.Ldc_r4, 0x00, 0x00, 0xC0, 0xFF,
+                (byte)ILOpCode.Ldc_r4, 0xCC, 0xCC, 0xCC, 0xCC,
                 (byte)ILOpCode.Ldc_r4, 0x00, 0x00, 0x80, 0xFF,
                 (byte)ILOpCode.Ldc_r4, 0x00, 0x00, 0x80, 0x7F,
                 (byte)ILOpCode.Ldc_r4, 0x01, 0x00, 0x00, 0x00
-            }, builder.ToArray());
+            };
+            // float.NaN may differ across architectures, in particular it's negative on x86 and positive elsewhere
```

Comment:
`float.NaN` is a constant defined in platform-neutral reference assembly. The platform-neutral reference assemblies should be exactly same regardless of the platform that they are built on.

It may be better to compensate for the non-deterministic NaN encoding in Roslyn. Roslyn should produce the exact same binary for given C# file, regardless of the platform that the binary is built on.

Reply:
If Roslyn isn't already doing this, it's likely a determinism bug and something they'd want to fix.

The JIT may also canonicalize in some cases, such as for constant folding.

Reply:
I fixed the issue with importing in #93285, so closing this PR.

I would still appreciate an answer to understand NaNs better.